conda-store-server/conda_store_server/action/generate_constructor_installer.py

```diff
@@ -0,0 +1,127 @@
+import os
+import pathlib
+import subprocess
+import sys
+import tempfile
+import warnings
+
+import yaml
+from conda_store_server import action, schema
+
+
+def get_installer_platform():
+    # This is how the default platform name is generated internally by
+    # constructor. For example: osx-arm64, linux-64, win-64.
+    # https://github.com/conda/constructor/blob/main/CONSTRUCT.md#available-platforms
+    from conda.base.context import context
+
+    return context.subdir
+
+
+@action.action
+def action_generate_constructor_installer(
+    context,
+    conda_command: str,
+    specification: schema.CondaSpecification,
+    installer_dir: pathlib.Path,
+    version: str,
+):
+    # Helpers
+    def print_cmd(cmd):
+        context.log.info(f"Running command: {' '.join(cmd)}")
+        context.log.info(
+            subprocess.check_output(cmd, stderr=subprocess.STDOUT, encoding="utf-8")
+        )
+
+    def write_file(filename, s):
+        with open(filename, "w") as f:
+            context.log.info(f"{filename}:\n{s}")
+            f.write(s)
+
+    # Checks if constructor is available
+    try:
+        command = [
+            "constructor",
+            "--help",
+        ]
+        print_cmd(command)
+    except FileNotFoundError:
+        warnings.warn(
+            "Installer generation requires constructor: https://github.com/conda/constructor"
+        )
+        return
```

Comment:
I now check whether constructor is available instead of creating a temporary environment and installing it every time.

constructor is part of the conda environment provided in the repo, so it'd be available in Docker, etc.